File: backend/app/schemas/master_data.py
"""
Master data schemas
"""

from pydantic import BaseModel


class StateResponse(BaseModel):
    """Schema for state response"""
    state_id: int
    state_name: str
    state_code: str
    
    class Config:
        from_attributes = True


class DistrictResponse(BaseModel):
    """Schema for district response"""
    district_id: int
    district_name: str
    district_code: str
    state_id: int
    
    class Config:
        from_attributes = True


class CollegeResponse(BaseModel):
    """Schema for college response"""
    college_id: int
    college_name: str
    state_id: int
    
    class Config:
        from_attributes = True


class CasteResponse(BaseModel):
    """Schema for caste response"""
    caste_id: int
    caste_name: str
    
    class Config:
        from_attributes = True


class QualificationResponse(BaseModel):
    """Schema for qualification response"""
    qualification_id: int
    qualification_name: str
    qual_code: int | None
    
    class Config:
        from_attributes = True


class StreamResponse(BaseModel):
    """Schema for stream response"""
    stream_id: int
    stream_name: str
    qual_code: int | None
    
    class Config:
        from_attributes = True

File: frontend/src/api/masterData.ts

import apiClient from './client'

export interface State {
  state_id: number
  state_name: string
  state_code: string
}

export interface District {
  district_id: number
  district_name: string
  district_code: string
  state_id: number
}

export interface College {
  college_id: number
  college_name: string
  state_id: number
}

export interface Caste {
  caste_id: number
  caste_name: string
}

export interface Qualification {
  qualification_id: number
  qualification_name: string
  qual_code: number | null
}

export interface Stream {
  stream_id: number
  stream_name: string
  qual_code: number | null
}

export const masterDataAPI = {
  getStates: async (): Promise<State[]> => {
    const response = await apiClient.get<State[]>('/master/states')
    return response.data
  },
  getDistricts: async (stateId: number): Promise<District[]> => {
    const response = await apiClient.get<District[]>(`/master/districts?state_id=${stateId}`)
    return response.data
  },
  getColleges: async (stateId: number): Promise<College[]> => {
    const response = await apiClient.get<College[]>(`/master/colleges?state_id=${stateId}`)
    return response.data
  },
  getCastes: async (): Promise<Caste[]> => {
    const response = await apiClient.get<Caste[]>('/master/castes')
    return response.data
  },
  getQualifications: async (): Promise<Qualification[]> => {
    const response = await apiClient.get<Qualification[]>('/master/qualifications')
    return response.data
  },
  getStreams: async (): Promise<Stream[]> => {
    const response = await apiClient.get<Stream[]>('/master/streams')
    return response.data
  },
}